docs/source/plugins.rst

```diff
@@ -0,0 +1,130 @@
+=======
+Plugins
+=======
+
+Rez is designed around the concept of plugins. Plugins can be used to extend rez's functionalities without modifying any of rez's source code.
+
+Plugins are currently bundled in the main rez repo, but will be split out
+to their own repos in the future.
+
+The built-in plugins are located at :gh-rez:`src/rezplugins`.
+
+Loaded plugins
+==============
+
+Currently loaded plugins can be queried by running ``rez -i``
+
+.. code-block:: console
+
+   $ rez -i
+
+   Rez 2.113.0
+
+   PLUGIN TYPE         NAME        DESCRIPTION                                        STATUS
+   -----------         ----        -----------                                        ------
+   build process       local       Builds packages on local host                      loaded
+   build process       remote      Builds packages on remote hosts                    loaded
+   build system        cmake       CMake-based build system                           loaded
+   build system        custom      Package-defined build command                      loaded
+   build system        make        Make-based build system                            loaded
+   package repository  filesystem  Filesystem-based package repository                loaded
+   package repository  memory      In-memory package repository                       loaded
+   release hook        amqp        Publishes a message to the broker.                 loaded
+   release hook        command     Executes pre- and post-release shell commands      loaded
+   release hook        emailer     Sends a post-release email                         loaded
+   release vcs         git         Git version control                                loaded
+   release vcs         hg          Mercurial version control                          loaded
+   release vcs         stub        Stub version control system, for testing purposes  loaded
+   release vcs         svn                                                            FAILED: No module named 'pysvn'
+   shell               cmd         Windows Command Prompt (DOS) shell.                loaded
+   shell               gitbash     Git Bash (for Windows) shell                       loaded
+   shell               powershell  Windows PowerShell 5                               loaded
+   shell               pwsh        PowerShell Core 6+                                 loaded
+
+
+Configuring plugins
+===================
+
+# TODO: Write this part.
```

Comment:
Tried poking through some code but AFAIK you can't actually configure plugins in your own `rezconfig.py` outside of the repo structure, but maybe I'm missing something? Can you help me out here @JeanChristopheMorinPerso?

The whole idea that you have to configure plugin settings in two different places in the repo, the "plugin type" settings at the root of the repo and then specific plugin settings in the `rezplugins` folder is pretty odd to me.

Reply:
You can configure plugins in your own rezconfig.py like so:

```python
plugins = {
    "filesystem": {}
}
```

Reply:
See https://github.com/AcademySoftwareFoundation/rez/blob/main/src%2Frez%2Fconfig.py#L643-L648